sdk/containerapps/Azure.ResourceManager.Applications.Containers/src/Generated/ManagedEnvironmentResource.cs

```diff
@@ -0,0 +1,599 @@
+        /// <summary>
+        /// Patches a Managed Environment using JSON Merge Patch
+        /// Request Path: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/managedEnvironments/{name}
+        /// Operation Id: ManagedEnvironments_Update
+        /// </summary>
+        /// <param name="waitUntil"> <see cref="WaitUntil.Completed"/> if the method should wait to return until the long-running operation has completed on the service; <see cref="WaitUntil.Started"/> if it should return after starting the operation. For more information on long-running operations, please see <see href="https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/LongRunningOperations.md"> Azure.Core Long-Running Operation samples</see>. </param>
+        /// <param name="data"> Configuration details of the Environment. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="data"/> is null. </exception>
+        public virtual ArmOperation Update(WaitUntil waitUntil, ManagedEnvironmentData data, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(data, nameof(data));
+
+            using var scope = _managedEnvironmentClientDiagnostics.CreateScope("ManagedEnvironmentResource.Update");
+            scope.Start();
+            try
+            {
+                var response = _managedEnvironmentRestClient.Update(Id.SubscriptionId, Id.ResourceGroupName, Id.Name, data, cancellationToken);
+                var operation = new ContainersArmOperation(_managedEnvironmentClientDiagnostics, Pipeline, _managedEnvironmentRestClient.CreateUpdateRequest(Id.SubscriptionId, Id.ResourceGroupName, Id.Name, data).Request, response, OperationFinalStateVia.Location);
+                if (waitUntil == WaitUntil.Completed)
+                    operation.WaitForCompletionResponse(cancellationToken);
+                return operation;
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Checks if resource name is available.
+        /// Request Path: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/managedEnvironments/{managedEnvironmentName}/checkNameAvailability
+        /// Operation Id: Namespaces_CheckNameAvailability
+        /// </summary>
+        /// <param name="content"> The check name availability request. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="content"/> is null. </exception>
+        public virtual async Task<Response<CheckNameAvailabilityResponse>> CheckNameAvailabilityNamespaceAsync(CheckNameAvailabilityContent content, CancellationToken cancellationToken = default)
```

Comment:
Can we rename the input and output parameter to avoid the duplications across RPs?
```suggestion
        public virtual async Task<Response<[RP or Resource]NameAvailabilityResult>> CheckNameAvailabilityNamespaceAsync([RP or Resource]NameAvailabilityContent content, CancellationToken cancellationToken = default)
```
Also, it's better to rename the method, because in other RPs, we use [this pattern](https://github.com/Azure/azure-sdk-for-net/blame/main/sdk/eventhub/Azure.ResourceManager.EventHubs/src/autorest.md#L18-L20).

Reply:
The `CheckNameAvailabilityResponse` and `CheckNameAvailabilityContent` are the shared models from common, so don't need to rename.
But yes the method name should be fixed

Reply:
Well, actually we removed these common models in `Azure.ResourceManager` and I have this proposal from the perspective of maintaining consistency. But it's ok to keep it unchanged for the beta.